src/client/datascience/interactive-window/interactiveWindow.ts

```diff
@@ -167,12 +167,28 @@ export class InteractiveWindow implements IInteractiveWindowLoadable {
             this.internalDisposables
         );
         this.internalDisposables.push(kernel);
-        await kernel.start();
+
```

Comment:
This function is rather confusing. Internal state is assigned to its result but it also updates that internal state in itself. 

What if you did this instead (and not assign the _kernelReadyPromise as the result of this function)

```
this._kernelReadyPromise = () => {
    try {
        await kernel.start();
        this.fileInKernel = undefined;
        await this.runIntialization(kernel, this.owner);
    } catch {
        // Do nothing
    }
};
```

Reply:
The exception could be thrown too if we want to make sure it comes out. Maybe like this instead:

```
let exception: Error | undefined;
const readyPromise = () => {
    try {
        await kernel.start();
        this.fileInKernel = undefined;
        await this.runIntialization(kernel, this.owner);
    } catch (e) {
        throw e;
        this._kernelReadyPromise = undefined
    }
};
this._kernelReadyPromise = readyPromise;
await readyPromise;
```

Reply:
Or maybe a simpler way, turn your 'ensureKernelReadyPromise' into a wrapper around 'createKernelReadyPromise' and have it behave like so:
```
public ensureKernelReadyPromise() {
    if (!this._kernelReadyPromise) {
        this._kernelReadyPromise = this.createKernelReadyPromise().catch(() => this._kernelReadyPromise = undefined);
    }
}

Reply:
And then replace all calls to 'createKernelReadyPromise' with the 'ensureKernelReadyPromise'. The error will still be thrown if somebody waits on the _kernelReadyPromise, but it won't prevent the next call from recreating it.